src/TransCreator/index.jsx

```diff
@@ -60,16 +61,23 @@ const TransCreator = ({ navigation }) => {
             setTransType={(type) => {
               setTransType(type);
               setTransLabel({});
-              setTransAmount(null);
+              setTransAmount('');
+              setExpStr('');
             }}
-
           />
-          <Item style={styles.dateItem}>
-            <DateSelector transDate={transDate} setTransDate={setTransDate} />
-          </Item>
-          <PageFooter createHandler={createHandler} />
         </View>
       </View>
+      <Keyboard
+        calculable={transLabel.name}
+        createHandler={createHandler}
+        updateCalDisplay={(val) => {
```

Comment:
since they are two different components, the keyboard shouldn't know about the calDisplay, let's call this onExpressionsChaneg and within the CalculatorKeyboard just ccall this method with an object like:
```js
{
  expression: '12 + 43',
  result: 55,
}
```
So only the CalculatorKeyboard component will depend on the calculator.js

Reply:
fixed